lib/manager/components/validation/MobilityDataValidationResult.js

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable no-fallthrough */
+import React, { useState } from 'react'
+import Icon from '@conveyal/woonerf/components/icon'
+import { ListGroupItem, Table } from 'react-bootstrap'
+
+import toSentenceCase from '../../../common/util/to-sentence-case'
+import {
+  mobilityDataValidationErrorMapping,
+  validationErrorIconLookup
+} from '../../util/version'
+
+import rules from './rules.json'
+
+// from https://stackoverflow.com/a/4149671
+function unCamelCase (s) {
+  return s
+    .split(/(?=[A-Z])/)
+    .join(' ')
+    .toLowerCase()
+}
+
+const NoticeTable = ({ headerOverides = {}, notices }) => {
+  if (notices.length === 0) return null
+
+  const headers = Object.keys(notices[0])
+
+  return (
+    <Table bordered className='table-fixed' fill hover striped>
+      <thead>
+        <tr>
+          {headers.map((header) => (
+            <th>
+              {headerOverides[header] || toSentenceCase(unCamelCase(header))}
+            </th>
+          ))}
+        </tr>
+      </thead>
+      <tbody>
+        {notices.map((notice) => (
+          <tr>
+            {headers.map((header, index) => {
+              const FieldWrapper =
+                (header === 'fieldValue' || header === 'message') ? 'pre' : React.Fragment
+
+              return (
+                <td key={`${header}-${index}`}>
+                  <FieldWrapper>{notice[header]}</FieldWrapper>
+                </td>
+              )
+            })}
+          </tr>
+        ))}
+      </tbody>
+    </Table>
+  )
+}
+
+// eslint-disable-next-line complexity
+const renderNoticeDetail = (notice) => {
+  switch (notice.code) {
+    case 'too_many_rows':
+      notice.csvRowNumber = notice.rowNumber
+    case 'fare_transfer_rule_duration_limit_type_without_duration_limit':
+    case 'fare_transfer_rule_duration_limit_without_type':
+    case 'fare_transfer_rule_missing_transfer_count':
+    case 'fare_transfer_rule_with_forbidden_transfer_count':
+      notice.filename = 'fare_transfer_rules.txt'
+    case 'empty_file':
+    case 'emtpy_row':
+    case 'missing_recommended_field':
+    case 'missing_timepoint_column':
+    case 'missing_required_file':
+    case 'missing_recommended_file':
+    case 'unknown_file':
+      return (
+        <ul style={{marginTop: '-10px'}}>
+          {notice.sampleNotices.map((notice) => (
+            <li>
+              {notice.filename}
+              {notice.csvRowNumber && `: row ${notice.csvRowNumber}`}
+            </li>
+          ))}
+        </ul>
+      )
+    default:
+      return (
+        <NoticeTable notices={notice.sampleNotices} />
+      )
+  }
+}
+
+const MobilityDataValidationResult = (props) => {
+  const { notice } = props
+  const rule = rules.find((rd) => rd.rule === notice.code)
+  if (!rule) return null
+
+  const errorClass = `gtfs-error-${mobilityDataValidationErrorMapping[notice.severity]}`
+  const [expanded, setExpanded] = useState(notice.totalNotices < 2)
```

Comment:
Not sure about this...

Reply:
What are you not sure about? If 2 is a good number for when to show the notices?

Reply:
Yes and if we should expand by default at all?